src/shared/components/modal/__tests__/ModalProvider.spec.tsx

```diff
@@ -0,0 +1,34 @@
+import * as React from 'react';
+import { Button } from '@patternfly/react-core';
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+import { createModalLauncher } from '../createModalLauncher';
+import { useModalLauncher, ModalProvider } from '../ModalProvider';
+
+describe('Modal Provider', () => {
+  it('should launch modal on button click', async () => {
+    const launchMyModal = createModalLauncher(() => <div>My Modal</div>, {
+      'data-testid': 'my-modal',
+    });
+    const TestComponent = () => {
+      const showModal = useModalLauncher();
+      return (
+        <Button data-testid="show-modal-button" onClick={() => showModal(launchMyModal({}))}>
```

Comment:
Test fails if you pass null to `launchMyModal`. The launcher should not assume its going to get an object everytime.